src/components/choose-locale/change-locale.component.tsx

```diff
@@ -0,0 +1,49 @@
+import { refetchCurrentUser } from "@openmrs/esm-api";
+import React, { useEffect, useState } from "react";
+import { updateUserProperties } from "./change-locale.resource";
+
+interface ChangeLocaleProps {
+  allowedLocales: Array<string>;
+  user: any;
+}
+
+export const ChangeLocale: React.FC<ChangeLocaleProps> = ({
+  allowedLocales,
+  user
+}) => {
+  const [userProps, setUserProps] = useState(user.userProperties);
+  const options = allowedLocales?.map(locale => (
+    <option value={locale} key={locale}>
+      {locale}
+    </option>
+  ));
```

Comment:
My preference would be to inline this in the `return` block

Reply:
I had this inlined but it was automatically formatted by some js plugin at pre-commit. Maybe 'js lint'?

Reply:
Sorry Samuel, when I say "inlined" I don't mean "in a single line," but rather that you should just put this whole value directly where the value goes (in the return block below). No temporary variable assignment. It's not important though.